app/controllers/api/v1/owners_controller.rb

```diff
@@ -54,4 +60,8 @@ def verify_gem_ownership
     return if @api_user.rubygems.find_by_name(params[:rubygem_id])
     render plain: "You do not have permission to manage this gem.", status: :unauthorized
   end
+
+  def verify_ownership_exists(owner)
```

Comment:
We had discussed that we will allow user to add owner again so that confirmation email can be sent (with rate limit). So, in `create` method, I use 
```ownership = Ownership.new_temporary_ownership(@rubygem, owner, @api_user)```

This finds if there is existing ownership or initializes new object. As it finds existing object, it doesn't throw an error in saving. So, it's not clear to the user. Also it sends the email again which can cause confusion.

This method is just to show a clear message to the user.
Any alternatives you suggest?

Reply:
Looking at the code is making me realize that it is getting complicated because we are trying to fit two responsibilities in one action.
Let's create a new action just for re-sending confirmation token. On the owner page under the confirmed column, we can show a link to resent confirmation mail if the user is unconfirmed.